Refresh balances on transaction

Currently the balance is not refreshed after a transaction. This could (should?) be done by listening to Transfer events.

Refreshing balance
meta-sender.component.ts:86 Error: Contract has not been deployed to detected network (network/artifact mismatch)
    at eval (contract.js:435)
    at ZoneDelegate.invoke (zone.js:388)
    at Zone.run (zone.js:138)
    at eval (zone.js:858)
    at ZoneDelegate.invokeTask (zone.js:421)
    at Zone.runTask (zone.js:188)
    at drainMicroTaskQueue (zone.js:595)
    at ZoneTask.invokeTask (zone.js:500)
    at ZoneTask.invoke (zone.js:485)
    at timer (zone.js:2025)

--- a/e2e/meta-sender.e2e-spec.ts
+++ b/e2e/meta-sender.e2e-spec.ts
@@ -20,7 +20,6 @@ describe('angular-truffle-box meta-sender', () => {
     page.clickSend();
 
     browser.driver.sleep(1000);
-    page.navigateTo();
 
     expect(page.getBalance()).toBe('9950');
     page.clickSelect();

--- a/src/app/meta/meta-sender/meta-sender.component.ts
+++ b/src/app/meta/meta-sender/meta-sender.component.ts
@@ -34,6 +34,14 @@ export class MetaSenderComponent implements OnInit {
     this.web3Service.artifactsToContract(metacoin_artifacts)
       .then((MetaCoinAbstraction) => {
         this.MetaCoin = MetaCoinAbstraction;
+        this.MetaCoin.deployed().then(deployed => {
+          console.log(deployed);
+          deployed.Transfer({}, (err, ev) => {
+            console.log('Transfer event came in, refreshing balance');
+            this.refreshBalance();
+          });
+        });
+
       });
   }
 

